site/plugins/propaganda/propaganda.py

```diff
@@ -51,56 +50,61 @@ def _gen_html(self, destination, images_folder, conf):
             item_path = os.path.join(images_folder, item)
             if os.path.isdir(item_path):
 
-                with open(os.path.join(item_path, 'url.txt'), 'r') as url_file:
+                with open(os.path.join(item_path, 'url.txt'), 'rt', encoding="utf8") as url_file:
                     title, url = url_file.readlines()
 
-                advertise = {
+                propaganda = {
                     'item': n,
                     'title': title.strip(),
                     'url': url.strip(),
-                    'image': os.path.join(item, 'logo_c.png')
+                    'image': os.path.join(item, conf['image_name'])
                 }
                 n += 1
-                items.append(list_item.format(**advertise))
-        conf["advertise_list"] = "\n".join(items)
+                items.append(list_item.format(**propaganda))
+        conf["propaganda_list"] = "\n".join(items)
 
         local_html = html
         for key, value in conf.items():
             local_html = local_html.replace("{{%s}}" % key, str(value))
 
-        with open(destination, "wb") as destination_file:
-            destination_file.write(local_html.encode('utf-8'))
-
+        with open(destination, "wt", encoding="utf8") as destination_file:
+            destination_file.write(local_html)
 
+    def _gen_html_task(self, images_folder, tasks, key):
```

Comment:
why not receive directly the key and the task, instead of doing tasks[key] everytime?